### PR TITLE
add BORG_CONFIG_DIR Fixes #3083

### LINKED ISSUE #3083: add BORG_CONFIG_DIR
from the mailing list:

> I've read them BORG_CACHE_DIR is OK but I see no way to relocate the ".config/borg" directory in its *entirety*!
>
> If I read the docs correctly, I can only relocate *parts* of ".config" with BORG_KEYS_DIR and BORG_SECURITY_DIR but there are other files besides these on the ".config/borg" directory.
>
> I'd really like a BORG_CONFIG_DIR variable to relocate the *whole* config dir. I want everything in ONE predictable place, especially stuff related to backups which is a really critical process in my PoV.

### PR DESCRIPTION
**Pull request by narendravardi**

### CODE REVIEW

File: src/borg/helpers/fs.py [hunk 1]
```diff
@@ -65,6 +65,15 @@ def get_cache_dir():
                 """).encode('ascii'))
     return cache_dir
 
+def get_config_dir():
```
Comment (ThomasWaldmann) [inline src/borg/helpers/fs.py]: did you run the tests and the pep8 checker (tox)?

File: src/borg/helpers/fs.py [hunk 2]
```diff
@@ -65,6 +65,15 @@ def get_cache_dir():
...
+    """Determine where to store whole config"""
+    xdg_config = os.environ.get('XDG_CONFIG_HOME', os.path.join(get_home_dir(), '.config'))
+    config_dir = os.environ.get('BORG_CONFIG_DIR', os.path.join(get_home_dir(xdg_config), 'borg'))
```
Comment (ThomasWaldmann) [inline src/borg/helpers/fs.py]: there is code in get_keys_dir and get_security_dir that could get removed by calling get_config_dir (currently it has no callers, so is not especially useful).

refactor them and run the tests afterwards.

**Comment by ThomasWaldmann** [comment]
don't merge master into your PR branch, rather rebase your PR on master.

**Comment by ThomasWaldmann** [comment]
hmm, you did not create a branch to work on (but you work on master). maybe read about git workflows.

**Comment by narendravardi** [comment]
@ThomasWaldmann, I can checkout to the [first commit](https://github.com/borgbackup/borg/pull/3206/commits/1fa23d162a562e147c348870a889a76decc0463e) of this PR to a new branch. Make changes (which are failing in the latest commit) and force push it to my repo.

But I am not sure if there are any side effects. To my knowledge, it might be possible I might have to create a new PR.

Also, can you please share the documentation on how to run tox in my local.

**Comment by ThomasWaldmann** [comment]
you need to bring your master branch to a unmodified state, so it corresponds to borg master.

then, from a master checkout, do a `git checkout -b feature-config-dir` to create a new branch and apply your changes there. push it to a corresponding branch in your github repo and make a PR from that.

**Comment by ThomasWaldmann** [comment]
about tox:

we have some developer docs in the borg docs and tox also has own docs.

but basically you just run `tox` from your workdir after you've setup everything.

**Comment by narendravardi** [comment]
@ThomasWaldmann, I have made changes in a new branch. That branch is behind of 14 commits from borgbackup/borg master branch.

Do you want to raise a new PR?

Link to new branch: https://github.com/narendravardi/borg/tree/add_borg_config_dir

**Comment by ThomasWaldmann** [comment]
yes, make a PR and then check what travis-ci says about it.

**Comment by ThomasWaldmann** [comment]
superceded by #3217

**Pull request closed without merging**